docs_src/security/Ch-Configuring-Add-On-Services.md

```diff
@@ -1 +1,189 @@
+secret tokens by default like those EdgeX core services and device-virtual service. There will be time of
+need to have EdgeX software stack generate the secret store token when end user wants to run the add-on
+services like `device-camera`, `app-http-export`, what have you in a secure mode.  
+
+In the following scenario, we assume the EdgeX services are running in Docker environments,
+and thus the examples are given in terms of Docker-compose ways.  It should not be much or bigger
+difference for `snap` running environment to apply the same steps or concepts if found to do so.
+
+If a user wants to configure and set up an add-on service, e.g. `device-camera`,
+he can do the steps of guidelines as follows to achieve that:
+
+## Make add-on services security-bootstrapper compatible
+
+To use the Docker entrypoint scripts of gating mechanism from `security-bootstrapper`,
+the Dockerfile of `device-camera` should inherit shell scripting capability like `alpine`-based
+as the base Docker image and should install `dumb-init`(see details in
+[Why you need an init system](https://github.com/Yelp/dumb-init#why-you-need-an-init-system))
+via `apk add --update` command.
+
+Dockerfile example using alpine-base image and add `dumb-init`:
+
+```dockerfile
+......
+FROM alpine:3.12
+
+# dumb-init needed for injected secure bootstrapping entrypoint script when run in secure mode.
+RUN apk add --update --no-cache dumb-init
+......
+
+```
+
+and then in the service itself should add `/edgex-init/ready_to_run_wait_install.sh` as the entrypoint script
+for the service in gating fashion and related Docker volumes from `edgex-init` and secretstore token which
+will be outlined in the next section.
+
+A good example of this will be like `app-serive-rules`:
+
+```yaml
+...
+  app-service-rules:
+    entrypoint: ["/edgex-init/ready_to_run_wait_install.sh"]
+    command: "/app-service-configurable ${DEFAULT_EDGEX_RUN_CMD_PARMS}"
+    volumes:
+      - edgex-init:/edgex-init:ro,z
+      - /tmp/edgex/secrets/app-rules-engine:/tmp/edgex/secrets/app-rules-engine:ro,z
+    depends_on:
+      - security-bootstrapper
+...
+
+```
+
+## Configure the secret store to use
+
+Make sure the TOML configuration file of add-on service like `device-camera` contains
+the proper `[SecretStore]` section.
+
+Example:
+
+```toml
+[SecretStore]
+Type = 'vault'
+Host = 'localhost'
+Port = 8200
+Path = 'my-service/'
+Protocol = 'http'
+RootCaCertPath = ''
+ServerName = ''
+TokenFile = '/tmp/edgex/secrets/my-service/secrets-token.json'
+  [SecretStore.Authentication]
+  AuthType = 'X-Vault-Token'
+```
+
+and then in the EdgeX service `secretstore-setup` environment section of `Docker-compose` file to add
+the service key or hostname to the environment variable `ADD_SECRETSTORE_TOKENS`:
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera'
```

Comment:
Should this be `my-service` to match the above instead of `device-camera`?

Reply:
good catch, i'll change `my-service` to `device-camera` to avoid conflicts and confusing.

Reply:
changed.